carma_ros2_utils/src/component_manager.cpp

```diff
@@ -281,16 +281,38 @@ ComponentManager::on_load_node(
             sev = RCUTILS_LOG_SEVERITY_WARN;
           }
           // Set the log level
-          auto result = rcutils_logging_set_logger_level(node_wrappers_[node_id].get_node_base_interface()->get_name(), sev);
-          
+
+          // Modify the node's fully qualified name to match the logger's name with namespaces
+          const char* name =  node_wrappers_[node_id].get_node_base_interface()->get_fully_qualified_name();
+
+          char modified_name[strlen(name) + 1]; // +1 for the null terminator
+
+          // Iterate through the original string, replacing '/' with '.' and skipping the first character if it's '/'
+          int j = 0;
+          for (int i = (name[0] == '/' ? 1 : 0); name[i] != '\0'; ++i) {
+              modified_name[j++] = (name[i] == '/' ? '.' : name[i]);
+          }
+          modified_name[j] = '\0'; // Null-terminate the modified string
+
+          auto result = rcutils_logging_set_logger_level(modified_name, sev);
```

Comment:
This can be simplified to
```cpp
std::string name{node_wrappers_[node_id].get_node_base_interface()->get_fully_qualified_name()};
std::replace(std::begin(name), std::end(name), '/', '.');

const auto result = rcutils_logging_set_logger_level(name.c_str(), sev);

Reply:
See comment below about storing `result`.